tests/system/python/packages/test_available_and_install_api.py

```diff
@@ -109,16 +110,48 @@ def test_available_service_packages(self, fledge_url):
         assert 'link' in jdoc
 
     def test_install_service_package(self, fledge_url):
+        pkg_name = "fledge-service-notification"
         conn = http.client.HTTPConnection(fledge_url)
-        data = {"format": "repository", "name": "fledge-service-notification"}
+        data = {"format": "repository", "name": pkg_name}
         conn.request("POST", '/fledge/service?action=install', json.dumps(data))
         r = conn.getresponse()
         assert 200 == r.status
         r = r.read().decode()
         jdoc = json.loads(r)
-        assert 'message' in jdoc
-        assert 'link' in jdoc
-        assert '{} is successfully installed'.format(data['name']) == jdoc['message']
+        assert 'id' in jdoc
+        assert '{} service installation started'.format(pkg_name) == jdoc['message']
+        assert jdoc['statusLink'].startswith('fledge/package/install/status?id=')
+
+        # Max retry count for to GET service package installed
+        max_retry_count = 5
+        while True:
+            if max_retry_count:
+                # GET Service Package Status
+                conn.request("GET", "/{}".format(jdoc['statusLink']))
+                r = conn.getresponse()
+                if r.status != 200:
+                    msg = "GET Service package status failed due to {} while attempting {}".format(
+                        r.reason, jdoc['statusLink'])
+                    print(msg)
+                    errors.append(msg)
+                    return
+                r = r.read().decode()
+                get_package_status_jdoc = json.loads(r)
+                if get_package_status_jdoc['packageStatus'][0]['status'] == "success":
+                    # Exit if SUCCESS
+                    break
+                elif get_package_status_jdoc['packageStatus'][0]['status'] == "failed":
+                    msg = "GET Service package status response failed while attempting {}".format(jdoc['statusLink'])
+                    print(msg)
+                    errors.append(msg)
+                    break
+                # sleep time added b/w retries
+                time.sleep(5)
+                max_retry_count -= 1
+            else:
+                msg = "{} installation failed. Timeout error!".format(pkg_name)
+                print(msg)
+                errors.append(msg)
```

Comment:
better to keep this as fixture `wait_for_pkg_install` and allow usage for any package and for any test? and  use existing max retry and wait fixtures